src/builtin.js

```diff
@@ -1151,34 +823,27 @@ Sk.builtin.filter = function filter (fun, iterable) {
     return ret(retval);
 };
 
-Sk.builtin.hasattr = function hasattr (obj, attr) {
-    Sk.builtin.pyCheckArgsLen("hasattr", arguments.length, 2, 2);
-    var special, ret;
-    if (!Sk.builtin.checkString(attr)) {
+Sk.builtin.hasattr = function hasattr(obj, pyName) {
+    if (!Sk.builtin.checkString(pyName)) {
         throw new Sk.builtin.TypeError("hasattr(): attribute name must be string");
     }
-
-    if (obj.tp$getattr) {
-        if (obj.tp$getattr(attr)) {
-            return Sk.builtin.bool.true$;
-        } else {
+    const res = obj.tp$getattr(pyName, true);
```

Comment:
I think this has to be surrounded with a Try/Catch to catch any AttributeError that might occur in a descriptor. This mirrors the behavior of CPython's implementation a little more closely: https://gist.github.com/taozle/f3b3d90337bfde97be68858ea89c66b7

At least, adding the following code seems to fix an issue with one of my weirder classes (which overrides every dunder method in Python).

```javascript
let res;
    try {
        res = obj.tp$getattr(pyName, true);
    } catch (e) {
        if (e instanceof Sk.builtin.AttributeError) {
            return false;
        } else {
            throw e;
        }
    }
```

Reply:
ta - curiosity did you override `__getattribute__`/`__getattr__` as dunder function with `Sk.builtin.func` or did you directly override `tp$getattr` with a js func?

I've added in the try catch and made it suspension aware.

Reply:
It's a [Python class](https://github.com/pedal-edu/pedal/blob/master/pedal/sandbox/result.py), not one written in JavaScript.